src/components/Content.tsx

```diff
@@ -43,7 +43,7 @@ const mapStateToProps = (state: State) => {
   const tutorialStep = isLoading ? tutorialStepLocal : +(getSetting(state, 'Tutorial Step') ?? 1)
 
   // do no sort here as the new object reference would cause a re-render even when the children have not changed
-  const rootThoughts = getThoughts(state, [ROOT_TOKEN])
+  const rootThoughts = getThoughts(state, [ROOT_TOKEN]).filter(({ value, rank }) => showHiddenThoughts || isChildVisible(state, [value], { value, rank }))
```

Comment:
I think you will have to `filter` within the `render` function. Otherwise `filter` creates a new object references and `mapStateToProps` will force a re-render even if the thoughts have not changed.

Reply:
It's also more efficient to check `showHiddenThought` outside the `filter` predicate since it is the same for all children.

Reply:
Agreed!